### PR TITLE
Include Diagnostics in Evolve Timer

### PR DESCRIPTION
**Pull request by ElisaRheaume**
Move reduced diags before evolve timing for more accurate elapsed time per loop iteration

### CODE REVIEW

**Comment by NeilZaim** [comment]
Thanks for this PR! I think CI might fail because `t_new` is used inside the diagnostics. You can maybe move this block below the `sync up time` block that updates `t_new` (and I would say also below the `warpx_py_afterstep` block) and see if it works better.

File: Source/Evolve/WarpXEvolve.cpp [hunk 1]
```diff
@@ -341,6 +326,21 @@ WarpX::Evolve (int numsteps)
             early_params_checked = true;
         }
 
+        // warpx_py_afterstep runs with the updated global time. It is included
+        // in the evolve timing.
+        if (warpx_py_afterstep) warpx_py_afterstep();
```
Comment (NeilZaim) [inline Source/Evolve/WarpXEvolve.cpp]: Maybe we want to keep this before the diagnostics, to keep the current code behavior?

File: Source/Evolve/WarpXEvolve.cpp [hunk 2]
```diff
@@ -341,6 +330,18 @@ WarpX::Evolve (int numsteps)
             early_params_checked = true;
         }
 
+        // create ending time stamp for calculating ellapsed time each iteration
```
Comment (NeilZaim) [inline Source/Evolve/WarpXEvolve.cpp]:
```suggestion
        // create ending time stamp for calculating elapsed time each iteration
```

**Review by NeilZaim — APPROVED** [review]
Thanks again for this PR, it looks good to me!

**Comment by dpgrote** [comment]
Would it be of interest to add a new separate timer for the diagnostics? Since the diagnostics can be a significant fraction of the run time, it would be useful to always have a timer for them so you know what fraction of the time step is spent on diagnostics.

**Comment by ax3l** [comment]
That's correct, but I would say the diagnostics should be well covered by the tiny profiler if we need that detail after a run.

**Review by ax3l — APPROVED** [review]
Thank you for the fix! :tada:

**Pull request merged**